src/pages/index.js

```diff
@@ -23,88 +37,225 @@ const profileDescriptionInput = document.querySelector(
 const cardForm = document.querySelector("#add-card-form");
 
-//Elements: Image Preview Modal
+//Elements: Image Preview Modal//
 const imagePreviewModal = new PopupWithImage({
   popupSelector: "#image-preview-modal",
 });
 
-//Elements: Form Validators
+// Elements: Delete Card Preview Modal//
+let currentCard;
+
+//Elements: Form Validators//
 const editFormValidator = new FormValidator(settings, profileEditForm);
 editFormValidator.enableValidation();
 
 const addFormValidator = new FormValidator(settings, cardForm);
 addFormValidator.enableValidation();
 
+const avatarFormValidator = new FormValidator(settings, avatarForm);
+avatarFormValidator.enableValidation();
+
 const toggleAddButtonState = () => addFormValidator.toggleButtonState();
 
-//Initialize Modals
-const editProfileModal = new PopupWithForm({
-  popupSelector: "#edit-profile-modal",
-  handleFormSubmit: (data) => {
-    userInfo.setUserInfo({ name: data.name, job: data.job });
-  },
-});
+///////////////////////////////////////////////////////////////////////////
+//       Initialize Classses & Modals       //////////////////////////////
+//////////////////////////////////////////////////////////////////////////
 
-const addCardModal = new PopupWithForm({
-  popupSelector: "#add-card-modal",
-  handleFormSubmit: (data) => {
-    const cardElement = createCard({ name: data.title, link: data.url });
-    section.addItem(cardElement);
-    cardForm.reset();
-    toggleAddButtonState();
+// Initialize API //
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c58369c4-dcd4-4208-8726-c10df21880b5",
+    "Content-Type": "application/json",
   },
 });
 
-// Initialize User Info
+// Initialize User Info //
 const userInfo = new UserInfo({
   nameSelector: ".profile__name",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__avatar",
 });
 
-// Intialize Section
+// Intialize Section //
 const section = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: (cardData) => {
+      //
       const cardElement = createCard(cardData);
+      //
       section.addItem(cardElement);
     },
   },
   ".cards"
 );
 
-section.renderItems();
+console.log("Section instance:", section);
 
-// Modal Event Listeners
-editProfileModal.setEventListeners();
-addCardModal.setEventListeners();
-imagePreviewModal.setEventListeners();
+// Fetch and Render Initial Data//
+api
+  .getAllData()
+  .then(({ user, cards }) => {
+    console.log("Initial user data:", user);
+    console.log("Initial cards data:", cards);
+    userInfo.setUserInfo(user);
+    section.renderItems(cards);
+  })
+  .catch((error) => console.error("Error loading data", error));
+
+const editAvatarModal = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: (data) => {
+    const userAvatar = {
+      avatar: data.avatar,
+    };
```

Comment:
You need to call `renderLoading` when you submit the avatar to show the loading effect